src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/AbstractCollectorTest.java

```diff
@@ -0,0 +1,39 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics;
+import java.time.Instant;
+import org.junit.Test;
+
+/**
+ * A base class that others can extend to validate their Metrics Collectors behavior
+ */
+public abstract class AbstractCollectorTest {
+  private PerformanceAnalyzerMetricsCollector uut;
+
+  // Implementors should call this in a setup() function to set their collector
+  public void setUut(PerformanceAnalyzerMetricsCollector collector) {
+    this.uut = collector;
+  }
+
+  // This is the only line that implementors need to modify
+  public abstract void validateMetric(String metric) throws Exception;
+
+  // This function does not need to be modified
+  @Test
+  public void validateMetrics() throws Exception {
+    uut.collectMetrics(Instant.now().toEpochMilli());
+    String metricString = uut.getValue().toString();
```

Comment:
I may not be understanding something correctly here, but don't we need an implementation of getValue in every collector? I was unable to locate one in the `GCInfoCollector`